docs/capabilities/event-emitter.md

```diff
@@ -0,0 +1,99 @@
+# Event Emitter
+
+The Event Emitter is a capability provided by AlgoKit Utils that allows for asynchronous event handling. It provides a flexible mechanism for emitting and listening to custom events, which can be particularly useful for debugging and extending functionality not available in the `algokit-utils-ts` package.
```

Comment:
```suggestion
The Event Emitter is a capability provided by AlgoKit Utils that allows for asynchronous event handling of lifecycle events. It provides a flexible mechanism for emitting and listening to custom events, which can be particularly useful for debugging and extending functionality not available in the `algokit-utils-ts` package.
```
My mental model for this is that we have a mechanism for emitting lifecycle type events that users can choose to handle.